content/languages/python/authoring/index.md

```diff
@@ -81,7 +81,56 @@ If a user solution is expected to use functions or classes from the `preloaded`
 
 Python kata use the [Codewars Python testing framework](/languages/python/codewars-test/) to implement and execute tests. You should read its reference page to find out how to use `describe` and `it` blocks for [organization and grouping](/languages/python/codewars-test/#grouping-tests), what [assertions](/languages/python/codewars-test/#assertions) are available, etc.
 
-You should notice that the Python testing framework produces one test output entry per assertion (and even 2 in [some special situations](/languages/python/codewars-test/#timeout-utility)), so the test output panel can get very noisy.
+
+#### Dynamically generated test cases
+
+It's possible to put functions decorated with `it` in a loop and use them as a construct similar to parametrized test cases known from other testing frameworks, for example:
+
+```python
+@test.describe("Generated test cases")
+def tests_with_generated_test_cases()
+    test_cases = generate_test_cases()
+    for msg, input, expected in test_case:
+        @test.it(msg)
+        def _():
+            actual = user_solution(input)
+            test.assert_equals(actual, expected)
+
+```
+
+This technique is liked by authors familiar with testing frameworks which provide parametrized or generated test cases out of the box, like NUnit, or JUnit. However, some caution is needed when this approach is used. Test suites organized this way can become large and can flood test output panel with many entries, making it unreadable or causing performance problems in client browsers.
+
+#### Decorated functions
+
+To create and present test output, Python testing framework uses parameters of `describe` and `it` decorators, and ignores actual names of decorated functions. Since the names are often redundant with titles of `describe` or `it` sections, they can be replaced with some placeholder name, for example `_`:
```

Comment:
```suggestion
To create and present test output, the Python testing framework uses parameters of `describe` and `it` decorators, and ignores actual names of decorated functions. Since the names are often redundant with titles of `describe` or `it` sections, they can be replaced with some placeholder name, for example, `_`:
```